MekHQ/src/mekhq/MekHQ.java

```diff
@@ -516,6 +527,90 @@ public void gamePhaseChange(GamePhaseChangeEvent e) {
         // Why Empty?
     }
 
+    // TODO: LUANA
+    public void autoResolveConcluded(AutoResolveConcludedEvent autoResolveConcludedEvent){
+        try {
+            String message = autoResolveConcludedEvent.controlledScenario() ?
+                "Your forces won the scenario. Did your side control the battlefield at the end of the scenario?" :
+                "Your forces lost the scenario. Do you want to declare your side as controlling the battlefield at the end of the scenario?";
+            String title = autoResolveConcludedEvent.controlledScenario() ? "Victory!" : "Defeat!";
+            boolean control = JOptionPane.showConfirmDialog(campaignGUI.getFrame(), message, title, JOptionPane.YES_NO_OPTION, JOptionPane.QUESTION_MESSAGE) == JOptionPane.YES_OPTION;
+            ResolveScenarioTracker tracker = new ResolveScenarioTracker(currentScenario, getCampaign(), control);
+            tracker.setClient(new AutoResolveClient(autoResolveConcludedEvent.getGame(), getCampaign().getPlayer()));
+            tracker.setEvent(autoResolveConcludedEvent);
+            tracker.processGame();
+
+            ResolveScenarioWizardDialog resolveDialog =
+                new ResolveScenarioWizardDialog(campaignGUI.getFrame(),
+                    true, tracker);
+            resolveDialog.setVisible(true);
+            if (resolveDialog.wasAborted()) {
+                for (UUID personId : tracker.getPeopleStatus().keySet()) {
+                    Person person = getCampaign().getPerson(personId);
+                    // person.setStatus(PersonnelStatus.ACTIVE);
+                    person.setHits(person.getHitsPrior());
+                }
+
+                // should I do something here? Maybe do the trigger?
+                // MekHQ.triggerEvent(new ScenarioResolvedEvent(currentScenario));
```

Comment:
I _think_ we want the trigger here, so it can be picked up by subscribers - though I don't know any that listen for this event, it's unlikely to hurt anything.